pkg/serving/config_changes.go

```diff
@@ -86,6 +86,27 @@ func UpdateImage(template *servingv1alpha1.RevisionTemplateSpec, image string) e
 	return nil
 }
 
+// UpdateContainerPort updates container with a give port
+func UpdateContainerPort(template *servingv1alpha1.RevisionTemplateSpec, port int32) error {
+	container, err := extractContainer(template)
+	if err != nil {
+		return err
+	}
+
+	if len(container.Ports) == 0 {
+		container.Ports = []corev1.ContainerPort{{
+			ContainerPort: port,
+		}}
+
+	} else {
+		for i, _ := range container.Ports {
```

Comment:
There should be only one. Please either assert that there is only one port (that is what Knative serving allows) and return an error or truncate to take only the first.

Tbh, I would just overwrite unconditionally any existing port.

Reply:
Overwritten the existing port